### PR TITLE
[compiler-rt] Update AllSupportedArchDefs.cmake for FreeBSD

### PR DESCRIPTION
**Pull request by arichardson**
This syncs the list of supported sanitizers with the matching code
in clang (`FreeBSD::getSupportedSanitizers()`),

### CODE REVIEW

**Comment by arichardson** [comment]
Ping

**Review by DimitryAndric — APPROVED** [review]
LGTM

File: compiler-rt/cmake/Modules/AllSupportedArchDefs.cmake [hunk 1]
```diff
@@ -59,20 +61,33 @@ else()
...
+
 set(ALL_HWASAN_SUPPORTED_ARCH ${X86_64} ${ARM64} ${RISCV64})
 set(ALL_MEMPROF_SUPPORTED_ARCH ${X86_64})
 set(ALL_PROFILE_SUPPORTED_ARCH ${X86} ${X86_64} ${ARM32} ${ARM64} ${PPC32} ${PPC64}
     ${MIPS32} ${MIPS64} ${S390X} ${SPARC} ${SPARCV9} ${HEXAGON}
     ${RISCV32} ${RISCV64} ${LOONGARCH64})
-set(ALL_TSAN_SUPPORTED_ARCH ${X86_64} ${MIPS64} ${ARM64} ${PPC64} ${S390X}
-    ${LOONGARCH64} ${RISCV64})
+if (OS_NAME MATCHES "FreeBSD")
+  set(ALL_TSAN_SUPPORTED_ARCH ${X86_64} ${MIPS64} ${ARM64})
```
Comment (brooksdavis) [inline compiler-rt/cmake/Modules/AllSupportedArchDefs.cmake:77]: Is the tsan case meant to retain MIPS64?

Comment (arichardson) [reply compiler-rt/cmake/Modules/AllSupportedArchDefs.cmake:77]: That's a good point, I just copied the values from FreeBSD::getSupportedSanitizers(), but MIPS64 is gone in recent FreeBSD, so might as well drop it here completely?

Comment (emaste) [reply compiler-rt/cmake/Modules/AllSupportedArchDefs.cmake:77]: I think so. mips64 is still supported in FreeBSD 13 which has about 2 more years before EOL, but I think it's quite likely that sanitizers are not being used/tested at all on FreeBSD/mips64.

**Review by emaste — APPROVED** [review]

**Pull request merged**